webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js

```diff
@@ -19,13 +19,17 @@ import { deleteSurvey, publishSurvey } from '@webapp/survey/actions'
 import { appModuleUri, homeModules } from '@webapp/app/appModules'
 import DeleteSurveyDialog from './components/deleteSurveyDialog'
 
-const SurveyInfo = props => {
-  const { surveyInfo, canEditDef, publishSurvey, deleteSurvey } = props
-
+const SurveyInfo = () => {
   const [showDeleteDialog, setShowDeleteDialog] = useState(false)
 
   const i18n = useI18n()
   const dispatch = useDispatch()
+  const history = useHistory()
+
+  const user = useSelector(AppState.getUser)
+  const surveyInfo = useSelector(SurveyState.getSurveyInfo)
+  const canEditDef = Authorizer.canEditSurvey(user, surveyInfo)
```

Comment:
and what about a ```useAuthCanEditSurvey```? defined in useAuth hook?

Reply:
ok